src/duplicacy_backupmanager.go

```diff
@@ -630,7 +634,9 @@ func (manager *BackupManager) Backup(top string, quickMode bool, threads int, ta
     }
     skippedFiles = append(skippedFiles, fileReader.SkippedFiles...)
 
-    manager.SnapshotManager.CleanSnapshotCache(localSnapshot, nil)
+    if !manager.config.dryRun {
```

Comment:
Glad you got this one.  CleanSnapshotCache is mainly to keep the last snapshot in the cache while deleting files from older snapshots, so we shouldn't really call it in dry-run mode.